trace/trace.go

```diff
@@ -43,6 +44,13 @@ func New(name string) *Trace {
 	return &Trace{name, time.Now(), nil}
 }
 
+// NewWithOutput creates a Trace with the specified name and klog output options defined
+func NewWithOutput(name string, writer io.Writer) *Trace {
+	// allows libraries to apply tracing and view output without declaring klog
+	klog.SetOutput(writer)
```

Comment:
The overall idea looks fine.
But I'm not sure if this is the best implementation - if someone will use this function, they will overwrite output of all klog calls? That seems a bit dangerous to me...

Instead of that, I would suggest refactoring it a bit so that we create a writer somehow connected to klog in New method or use the ncoming from NewWithOutput instead of creating, so that the impact is more local.

Thoughts?
@dims ?

Reply:
I'm unclear whether calling `klog.setOutput` actually flushes or clears all thats been [collected](https://github.com/kubernetes/klog/blob/master/klog.go#L738). I didn't want to impose user facing changes with in mucking with `New`. 

Waiting for more direction on this before refactoring 👍

Reply:
> I'm unclear whether calling klog.setOutput actually flushes or clears all thats been collected. 

I didn't only mean about things that happened already. I was more afraid of what will happen if I use klog later in some independent code.

> I didn't want to impose user facing changes by mucking with New.

I wasn't saying about user-facing changes - I was only suggesting some implementation change underneath.

Reply:
@wojtek-t yes I see your point. I didn't think that far ahead with the change. Do you have any suggestions for updating the PR? 

@dims thoughts on this?

Reply:
It requires understanding it a bit better, so I don't have any specific suggestions TBH.